Helper/Api/Products.php

```diff
@@ -771,24 +776,21 @@ public function update($storeId, $target = null, array $result = [])
             }
 
             $sourceItemData = [
-                'source_code' => 'default',
+                'source_code' => $this->configHelper->getStockSource(),
                 'status' => 1
             ];
 
             $shouldUpdateStock = false;
             $stockItem = $this->stockItem->load($this->productRepository->get($sku)->getId(), 'product_id');
 
-            (float) $product_stock_value = $product['product_stock']['value'];
+            (float) $product_stock_value = (array_key_exists('orderable_stock_value', $result)) ? $result['orderable_stock_value'] : null;
             $product_stock_unlimited = $product['product_stock']['unlimited'];
+            $backorder_enabled = (array_key_exists('backorder_enabled', $result)) ? $result['backorder_enabled'] : null;
 
-            if ($product_stock_unlimited &&
-                ($stockItem->getBackorders() == false || $stockItem->getUseConfigBackOrders() == true)
-            ) {
+            if ($backorder_enabled === true) {
```

Comment:
those changes do not look correct, unlimited is not used and it should. Can you extract it to a function and unittest it (see the [unit test for it in woocomerce](https://github.com/storekeeper-company/storekeeper-woocommerce-b2c/blob/4c48e4dc069128260f256cd742fe2d405408ceb2/tests/unit/Imports/ProductImportTest.php#L159) ?

that [how it is in woocomerce 
](https://github.com/storekeeper-company/storekeeper-woocommerce-b2c/blob/master/src/StoreKeeper/WooCommerce/B2C/Imports/AbstractProductImport.php#L214)
```php
        $backorder_enabled = $dot->get('backorder_enabled', false);
        $unlimited_stock = $dot->get($this->cleanDotPath($stock_path.'.unlimited'));
        $orderable_stock_quantity = $dot->get($this->cleanDotPath($shop_product_path.'.orderable_stock_value'));

        $in_stock = null === $orderable_stock_quantity || $orderable_stock_quantity > 0;
        $manage_stock = !$unlimited_stock;
        $stock_quantity = $manage_stock ? $orderable_stock_quantity : null;

        if (!is_null($stock_quantity) && $stock_quantity < 0) {
            $stock_quantity = 0;
        }

        $stock_status = self::STOCK_STATUS_IN_STOCK;
        if (!$in_stock) {
            if ($backorder_enabled && !$manage_stock) {
                $stock_status = self::STOCK_STATUS_ON_BACKORDER;
            } else {
                $stock_status = self::STOCK_STATUS_OUT_OF_STOCK;
            }
        }

        return [$manage_stock, $stock_quantity, $stock_status];
        
```